finagle-http/src/main/scala/com/twitter/finagle/Http.scala

```diff
@@ -107,6 +107,7 @@ object Http extends Client[Request, Response] with HttpRichClient
     val stack: Stack[ServiceFactory[Request, Response]] =
       StackClient.newStack
         .replace(TraceInitializerFilter.role, new HttpClientTraceInitializer[Request, Response])
+        .replace(StackClient.Role.prepFactory, DelayedRelease.module)
```

Comment:
Right, the relevant (to this issue) difference between Stack and ClientBuilder clients is that ClientBuilder has access to the codec, via which it can inject codec-specific logic as prepConn and prepFactory. In the HTTP codec, this is implemented as DelayedReleaseService wrapping.